code/game/machinery/big_manipulator.dm

```diff
@@ -0,0 +1,282 @@
+/// Manipulator Core. Main part of the mechanism that carries out the entire process.
+/obj/machinery/big_manipulator
+	name = "Big Manipulator"
+	desc = "Take and drop objects. Innovation..."
+	icon = 'icons/obj/machines/big_manipulator_parts/big_manipulator_core.dmi'
+	icon_state = "core"
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/big_manipulator
+	greyscale_colors = "#d8ce13"
+	greyscale_config = /datum/greyscale_config/big_manipulator
+	/// How many time manipulator need to take and drop item.
+	var/working_speed = 2 SECONDS
+	/// Using high tier manipulators speeds up big manipulator and requires more energy.
+	var/power_use_lvl = 0.2
+	/// When manipulator already working with item inside he don't take any new items.
+	var/on_work = FALSE
+	/// Activate mechanism.
+	var/on = FALSE
+	/// Dir to get turf where we take items.
+	var/take_here = NORTH
+	/// Dir to get turf where we drop items.
+	var/drop_here = SOUTH
+	/// Turf where we take items.
+	var/turf/take_turf
+	/// Turf where we drop items.
+	var/turf/drop_turf
+	/// Obj inside manipulator.
+	var/datum/weakref/containment_obj
+	/// Other manipulator component.
+	var/obj/effect/manipulator_hand
+
+/obj/machinery/big_manipulator/Initialize(mapload)
+	. = ..()
+	take_turf = get_step(src, take_here)
+	drop_turf = get_step(src, drop_here)
+	create_manipulator_hand()
+	RegisterSignal(manipulator_hand, COMSIG_QDELETING, PROC_REF(on_hand_qdel))
+	manipulator_lvl()
+	if(on)
+		press_on(FALSE)
```

Comment:
```suggestion
		press_on(pressed_by = null)
```
If it's not immediately clear what your parameter is, named args are preferred.
Also pressed_by makes me thing you expect a person or null, FALSE is for bools